File: Preproc/simple_BIO.pl
#!/usr/bin/perl -w

use strict;

use vars qw ();

my $data_d = $ARGV[0];
my $in_d = "$data_d/conll";
my $out_d = "$data_d/bio";

if (not -e $out_d) {
    system("mkdir $out_d");
}

opendir(D,$in_d)||die;
my @File = grep /\w/,readdir D;
closedir(D);

foreach my $f (@File) {
    open(I,"$in_d/$f")||die;
    open(O,">$out_d/$f")||die;
    while(<I>) {
	chomp;
	my @F = split(/\t/,$_);
	if (/^([BI])\-Mutation/) {
	    print O "$F[3]\t$1\n";
	}
	elsif ($_ ne "") {
	    print O "$F[3]\tO\n";
	}
	else {
	    print O "$_\n";
	}
    }
    close(I);
    close(O);
}

File: Preproc_py/simple_BIO.py
import os,sys,glob
import re

data_d = sys.argv[1]
in_d = data_d + "/conll/"
out_d = data_d + "/bio/"

if not os.path.isdir(out_d):
    os.mkdir(out_d)

os.chdir(in_d)
Files = glob.glob("*")

for f in Files:
    in_f = open(in_d + f,'r')
    out_f = open(out_d + f,'w')
    for line in in_f:
        line = line.rstrip()
        L = line.split("\t")
        m = re.search("^([BI])-Mutation",line)
        if m:
            out_f.write(L[3]+"\t"+m.group(1)+"\n")
        elif not line == "":
            out_f.write(L[3]+"\tO\n")
        else:
            out_f.write(line+"\n")
    in_f.close()
    out_f.close()

